### PR TITLE
Issue/2808 convert toes6 classes

### PR DESCRIPTION
**Pull request by eleanor-heath**
https://github.com/adaptlearning/adapt_framework/issues/2808

### CODE REVIEW

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

File: js/adapt-languagePicker.js [hunk 1]
```diff
@@ -1,85 +1,80 @@
...
-    if (!Adapt.config.get('_languagePicker')._isEnabled) return;
+/**
+ * Once the Adapt config has loaded, check to see if the language picker is enabled. If it is:
+ * - stop the rest of the .json from loading
+ * - set up the language picker model
+ * - register for events to allow us to display the language picker icon in the navbar on pages and menus
+ * - wait for offline storage to be ready so that we can check to see if there's a stored language choice or not
+ */
+function onConfigLoaded() {
+  if (!Adapt.config.has('_languagePicker') || !Adapt.config.get('_languagePicker')._isEnabled) return;
```
[the remark this answers is not in the document]
Comment (tomgreenfield) [reply js/adapt-languagePicker.js]: This needs to match the suggestion and be `.get()`.

File: js/languagePickerModel.js [hunk 2]
```diff
@@ -1,123 +1,121 @@
...
+      blocks: []
+    });
+  }
+
+  getTrackableState() {
+    const components = _.compact(this.getState(Adapt.components.models));
+    const blocks = _.compact(this.getState(Adapt.blocks.models));
+    return {
+      components,
+      blocks
```
[the remark this answers is not in the document]
Comment (tomgreenfield) [reply js/languagePickerModel.js]: Prefer native functions over Underscore when they don't provide any readability benefits.

**Comment by eleanor-heath** [comment]
@tomgreenfield

> Could also make the code in adapt-languagePicker.js into a controller.

How would I go about doing this?

**Comment by oliverfoster** [comment]
Same as you did for resources https://github.com/adaptlearning/adapt-contrib-resources/blob/master/js/adapt-contrib-resources.js#L5

**Comment by tomgreenfield** [comment]
Here's a very simple example: https://github.com/adaptlearning/adapt_framework/blob/master/src/core/js/headings.js. It sets up its own listeners in the initialize function.

**Comment by eleanor-heath** [comment]
@oliverfoster @tomgreenfield thank you both :) Ive pushed up a new commit to include the language picker controller.

File: js/adapt-languagePicker.js [hunk 3]
```diff
@@ -1,44 +1,46 @@
...
-  './languagePickerNavView',
-  './languagePickerModel'
-], function(Adapt, LanguagePickerView, LanguagePickerNavView, LanguagePickerModel) {
+import Adapt from 'core/js/adapt';
+import LanguagePickerView from './languagePickerView';
+import LanguagePickerNavView from './languagePickerNavView';
+import LanguagePickerModel from './languagePickerModel';
 
-  var languagePickerModel;
+let languagePickerModel;
```
Comment (tomgreenfield) [inline js/adapt-languagePicker.js:6]: This can be a property of the controller rather than a global.

File: js/adapt-languagePicker.js [hunk 4]
```diff
@@ -1,44 +1,46 @@
...
+    const config = Adapt.config.get('_languagePicker');
+    if (!config?._isEnabled) return;
 
     Adapt.config.set('_canLoadData', false);
 
-    languagePickerModel = new LanguagePickerModel(Adapt.config.get('_languagePicker'));
+    languagePickerModel = new LanguagePickerModel(config);
 
-    Adapt.on('router:menu router:page', setupNavigationView);
+    Adapt.on('router:menu router:page', this.setupNavigationView);
```
Comment (tomgreenfield) [inline js/adapt-languagePicker.js:29]: Here and on line 35, the controller can do its own listening, i.e. `this.listenTo()`/`this.listenToOnce()`.

File: js/adapt-languagePicker.js [hunk 5]
```diff
@@ -3,12 +3,14 @@ import LanguagePickerView from './languagePickerView';
...
-let languagePickerModel;
-
 class LanguagePicker extends Backbone.Controller {
   
   initialize() {
-    this.listenTo(Adapt, 'configModel:dataLoaded', this.onConfigLoaded);
+    this.listenTo(Adapt, {
+      'configModel:dataLoaded': this.onConfigLoaded,
+      'router:menu router:page': this.setupNavigationView,
+      'offlineStorage:ready': this.onOfflineStorageReady
```
Comment (tomgreenfield) [inline js/adapt-languagePicker.js:12]: Apologies if I wasn't very clear – the locations of the listeners were fine – it was just that the `LanguagePicker` class should be controlling the event  i.e. `this.listenTo()`/`this.listenToOnce()`.

We don't want the handlers firing if Language Picker isn't enabled so the listener setups should be moved back to lines 29 & 35 respectively.

**Review by tomgreenfield — APPROVED** [review]

**Review by oliverfoster — APPROVED** [review]

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by ethan-lp — APPROVED** [review]
👀

**Comment by oliverfoster** [comment]
The behaviour from Cahir's pr needs to get merged.

**Pull request merged**